### PR TITLE
chart analysis type utils

### PR DESCRIPTION
**Pull request by rickydoar**
added logarithmic and rolling data transformation utils

### CODE REVIEW

**Comment by tdumitrescu** [comment]
IRB

**Comment by rickydoar** [comment]
Changed to match IRB rolling average. Logarithmic is done by Highcharts.

**Comment by tdumitrescu** [comment]
Is this exactly the same as in IRB? what can/should we extract to common as a result?

**Comment by rickydoar** [comment]
The logic is identical. I changed variable names slightly so it was more readable as to what was happening.

**Comment by tdumitrescu** [comment]
So now we have two almost-but-not-quite-the-same versions?

**Comment by rickydoar** [comment]
I would assume IRB would start using the mp-common utils here, change nothing else about their code and then we'd have only one version.

**Comment by tdumitrescu** [comment]
That's fine - can you open an IRB PR to swap it out?

**Comment by rickydoar** [comment]
The functions are identical with `window` => `values` and `windowSize` => `rollingCount`, since the previous variable names didn't seem very intuitive. I can change it back if you feel strongly about it?

**Comment by rickydoar** [comment]
Yup can do. Can I do it after this one is merged?

File: lib/util/index.js [hunk 1]
```diff
@@ -180,3 +180,57 @@ export function truncateMiddle(string, len) {
...
+    case `day`:
+      return 7;
+    case `week`:
+      return 5;
+    case `month`:
+      return 3;
+  }
+}
+
+export function mapObjectKeys(obj, callback) {
```
Comment (tdumitrescu) [inline lib/util/index.js]: is this different from https://github.com/mixpanel/mixpanel-common/blob/master/lib/util/index.js#L103-L106?

**Comment by tdumitrescu** [comment]
Sure but how about before that you also copy the IRB tests for `nestedObjectRolling` so we can have some confidence that they behave the same?

File: lib/util/index.js (hunk 1 above)
Comment (tdumitrescu) [inline lib/util/index.js]: or uhh was it actually this one https://github.com/mixpanel/mixpanel-common/blob/master/lib/util/index.js#L108-L111

Comment (rickydoar) [inline lib/util/index.js]: It is different, but it's not different than https://github.com/mixpanel/mixpanel-common/blob/master/lib/util/index.js#L108-L111 😄

Comment (tdumitrescu) [inline lib/util/index.js]: 4 minutes too late homie

Comment (rickydoar) [inline lib/util/index.js]: hah got me

**Pull request merged**